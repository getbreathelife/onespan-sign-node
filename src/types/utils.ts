/**
 * Make `T` either its own type or null
 * @public
 */
export type Nullable<T> = T | null;

/**
 * Recursively make properties in `T` optional
 * @public
 */
export type RecursivePartial<T> = {
  [P in keyof T]?: RecursivePartial<T[P]>;
};
