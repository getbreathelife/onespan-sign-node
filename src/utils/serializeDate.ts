export function serializeDate(date: string | Date | undefined): string | undefined {
  if (date instanceof Date) {
    return date.toISOString();
  }

  if (typeof date === 'string' || typeof date === 'undefined') {
    return date;
  }

  throw new Error('Invalid parameter provided! Expect to be an instance of Date, string or undefined');
}
