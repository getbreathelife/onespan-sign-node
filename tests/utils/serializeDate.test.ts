import { serializeDate } from '../../src/utils/serializeDate';

describe('serializeDate', () => {
  it.each`
    type           | value
    ${'string'}    | ${'test123'}
    ${'undefined'} | ${undefined}
  `('passes $type parameter through untouched', ({ value }) => {
    expect(serializeDate(value)).toBe(value);
  });

  it('returns ISO date string when the parameter is a Date object', () => {
    const date = new Date(1641853087684);
    expect(serializeDate(date)).toEqual(date.toISOString());
  });
});
