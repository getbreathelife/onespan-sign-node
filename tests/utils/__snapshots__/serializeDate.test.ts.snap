// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializeDate throws an error if the provided parameter is an invalid type 1`] = `"Invalid parameter provided! Expect to be an instance of Date, string or undefined"`;
